openselery/openselery.py

```diff
@@ -175,99 +193,141 @@ def gather(self):
+            # calc release weights
             self.log("Add additional weight to release contributors of last " +
                     str(self.config.releases_included)+" releases")
             # Create a unique list of all release contributor
             release_contributor = git_utils.find_release_contributor(
-                local_repo, self.config.releases_included)
+                self.config.directory, self.config.releases_included)
             release_contributor = set(i.lower() for i in release_contributor)
             self.log("Found release contributor: "+str(len(release_contributor)))
-            for idx,user in enumerate(contributor):
+            for idx,user in enumerate(mainContributors):
                 if user.stats.author.email.lower() in release_contributor:
                     release_weights[idx]=self.config.release_weight
-                    self.log("Github email address matches git email from last release: " +user.stats.author.name )
+                    self.log("Github email matches git commit email of release of contributor: " +user.stats.author.login )
             self.log("Release Weights:" +str(release_weights))
             # considers all release contributor equal
             release_contributor = set(release_contributor)
 
-        # create uniform probability
-        self.log("Start with unifrom porbability weights for contributors")
-        uniform_weights = selery_utils.calculateContributorWeights(
-            contributor, self.config.uniform_weight)
-        self.log("Uniform Weights:" +str(uniform_weights))
-
         # sum up the two list with the same size
-        total_weights = [x + y for x, y in zip(uniform_weights, release_weights)]
+        combined_weights = [x + y for x, y in zip(uniform_weights, release_weights)]
 
-        self.log("Total Weights:" +str(total_weights))
+        self.log("Total Weights:" +str(combined_weights))
         # read @user from commit
-        return total_weights
+        return combined_weights, mainContributors
 
-    def choose(self, contributors, repo_path, weights):
+    def split(self, contributors, weights):
         recipients = []
-
+        
         # chose contributors for payout
         self.log("Choosing recipients for payout")
         if len(contributors) < 1:
             self.logError("Could not find any contributors to payoff")
             raise Exception("Aborting")
 
-        recipients = random.choices(
-            contributors, weights, k=self.config.number_payout_contributors_per_run)
-        for contributor in recipients:
-            self.log(" -- '%s': '%s' [w: %s]" % (contributor.stats.author.html_url,
-                                              contributor.stats.author.name, weights[contributors.index(contributor)]))
-            self.log("  > via project '%s'" % contributor.fromProject)
-        return recipients
+        if self.config.random_split and not self.config.full_split:
```

Comment:
@cornerman already said it but ill repeat it here:

the splitting / choosing of contributors should be based on a single defined keyword regarding a single obvious rule:
`self.config.split_mode` therefore should be a string containing the "action" to take, so `random`, `full`, `custom`, `main_only`, and so on ...